### DISCUSSION TITLE
React "react-scripts start" fails in new Codespace

### DISCUSSION THREAD

**Discussion by edwinhollen** [community/community#26365] [Codespaces]
<p>Hi all, I’m running into a <strong>reproducible</strong> issue when trying to run a newly created React app. I can reproduce this on a new repository:</p>
<ol>
<li>Create a new repo with a README.</li>
<li>Start a Codespace on the <code>main</code> branch.</li>
<li>Run <code>npx create-react-app .</code>
</li>
<li>Run <code>yarn start</code>
</li>
<li>Fails with this error:</li>
</ol>
<pre><code class="lang-auto">Starting the development server...

events.js:287
      throw er; // Unhandled 'error' event
      ^

Error: spawn /home/codespace/.vscode-remote/bin/caf01baa45b9816e2e1d8a2de22b40cf89459c03/bin/code-insiders --openExternal ENOENT
    at Process.ChildProcess._handle.onexit (internal/child_process.js:267:19)
    at onErrorNT (internal/child_process.js:469:16)
    at processTicksAndRejections (internal/process/task_queues.js:84:21)
Emitted 'error' event on ChildProcess instance at:
    at Process.ChildProcess._handle.onexit (internal/child_process.js:273:12)
    at onErrorNT (internal/child_process.js:469:16)
    at processTicksAndRejections (internal/process/task_queues.js:84:21) {
  errno: 'ENOENT',
  code: 'ENOENT',
  syscall: 'spawn /home/codespace/.vscode-remote/bin/caf01baa45b9816e2e1d8a2de22b40cf89459c03/bin/code-insiders --openExternal',
  path: '/home/codespace/.vscode-remote/bin/caf01baa45b9816e2e1d8a2de22b40cf89459c03/bin/code-insiders --openExternal',
  spawnargs: [ 'http://localhost:3000' ]
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
</code></pre>
<p>And some versions for reference:</p>
<p><code>npm --version 6.14.4</code></p>
<p><code>node --version v12.16.3</code></p>
<p><code>"react": "^17.0.1",     "react-dom": "^17.0.1",     "react-scripts": "4.0.0"</code></p>
<p>FYI on my local machine (Mac OS) this runs normally; I’m only experiencing this in a Codespace.</p>
<p>Anyone else experiencing this? Any ideas what’s going on? :sweat_smile:</p>

**Comment by edwinhollen** [comment]
<p>Update: I discovered a workaround:</p>
<p><code>sudo yarn start</code></p>
<p>This starts the React development server normally in the Codespace.</p>

**Comment by jkeech** [comment, marked as answer]
<p>It looks like the <code>react-dev-utils</code> package is trying to open the browser from the remote container, which is failing. In Codespaces, we really want to launch the browser on the client side, not the remote side. If you first run <code>export BROWSER=none</code> before running <code>yarn start</code>, it works since it skips trying to launch the browser on the remote side.</p>
<p>We’ll think through how we can make this scenario work better out of the box.</p>

**Comment by edwinhollen** [comment]
<p>Thanks John, I will use that in my start script! :+1: Much better solution than sudo. In addition, I noticed that a <code>.env</code> file at the project root containing a <code>BROWSER=none</code> entry doesn’t seem to get picked up inside the Codespaces environment; not sure if that’s related, but might be a good note for people setting up React in there.</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://create-react-app.dev/img/favicon/favicon.ico" class="site-icon" width="64" height="64">
      <a href="https://create-react-app.dev/docs/advanced-configuration/" target="_blank" rel="noopener nofollow ugc">create-react-app.dev</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/362;"><img src="https://user-images.githubusercontent.com/950704/181097347-2564f4cc-8076-4c57-a72a-16c50c4f049f.png" class="thumbnail" width="690" height="362" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/e/e9d52951b6799822784160951f379815293e6f9a_2_10x10.png"></div>

<h3><a href="https://create-react-app.dev/docs/advanced-configuration/" target="_blank" rel="noopener nofollow ugc">Advanced Configuration | Create React App</a></h3>

<p>You can adjust various development and production settings by setting environment variables in your shell or with .env.</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by jkeech** [comment]
<p><a class="mention" href="/edwinhollen">@edwinhollen</a>, following up, we’ve fixed this so that it works like you’d expect without having to disable the browser. You can remove the <code>BROWSER=none</code> workaround now. You’ll notice that the client browser is opened automatically when you run <code>yarn start</code> for this scenario.</p>
<p>Internally, you’ll notice that VS Code is now setting the <code>BROWSER</code> variable so that it can intercept requests to open the browser from the remote side and then replay them on the client side.</p>

**Comment by edwinhollen** [comment]
<p>Amazing, thank you for investigating John, and I’m a big fan of Codespaces :slight_smile: Keep up the great work!</p>
